### PR TITLE
Combine dependabot upgrades to avoid go.sum thrashing

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

Combines the following PRs to avoid go.sum thrashing on main
* ~https://github.com/DataDog/datadog-agent/pull/22792~ (reverted: build errors)
* ~https://github.com/DataDog/datadog-agent/pull/22790~ (reverted: build errors)
* https://github.com/DataDog/datadog-agent/pull/22791
* https://github.com/DataDog/datadog-agent/pull/22789

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

File: go.mod
Comment (pgimalac) [inline go.mod:1]: Looks like the otel update is breaking build 😕

Comment (jeremy-hanna) [reply go.mod:1]: yeah it does not look to be compatible with some submodule imports

```
--- Setting rtloader paths to lib:/Users/jeremy.hanna/go/src/github.com/DataDog/datadog-agent/dev/lib | header:/Users/jeremy.hanna/go/src/github.com/DataDog/datadog-agent/dev/include | common headers:
# go.opentelemetry.io/collector/service/internal/status
../../go/pkg/mod/go.opentelemetry.io/collector/service@v0.91.0/internal/status/status.go:162:13: undefined: component.StatusFunc
# go.opentelemetry.io/collector/internal/sharedcomponent
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:37:24: c.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:38:16: c.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:39:30: telemetrySettings.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:93:19: r.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:95:20: r.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:109:19: r.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:112:20: r.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
../../go/pkg/mod/go.opentelemetry.io/collector@v0.91.0/internal/sharedcomponent/sharedcomponent.go:114:20: r.telemetry.ReportComponentStatus undefined (type *component.TelemetrySettings has no field or method ReportComponentStatus)
```

```
$ go mod graph | grep collector | grep 91.0
...
github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter@v0.0.0-00010101000000-000000000000 go.opentelemetry.io/collector@v0.91.0
```

looks to need coordination with `exporter/serializerexporter` which utilizes a pinned collector/consumer version but that upgrade is not trivial

Comment (jeremy-hanna) [reply go.mod:1]: reverting the otel dependabot upgrades: https://github.com/DataDog/datadog-agent/pull/22819/commits/d8d5c168ec5225b5c4926edca9011312a164a98a

**Review by pgimalac — APPROVED** [review]

**Review by rayz — APPROVED** [review]

**Pull request merged**